server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -168,16 +176,24 @@ public int getNumOpenConnections()
       final QueryMetrics<? super Query<T>> queryMetrics = toolChest.makeMetrics(query);
       queryMetrics.server(host);
 
+      long timeoutAt = ((Long) context.get(QUERY_FAIL_TIME)).longValue();
```

Comment:
What I'm saying is that I would consider the whole concept of queries making other queries as not officially supported. If you want to do that in an extension then go for it but there is no official API for it.